fitbenchmarking/parsing/cutest_parser.py

```diff
@@ -0,0 +1,240 @@
+"""
+This file calls the pycutest interface for SIF data
```

Comment:
So from my memories of our weekly meetings, the complexity of the cutest parser is that in a lot of instances you need to write need files, which explains what's going on with the `TemporaryDirectory` and the writing in `_setup_data`.  I would recommend that you add something about this at the module level, as this is a specific quirk of the cutest parser that it would be useful to document - I imagine anyone with no cutest knowledge who has to work on this at a later date will thank you for it!

Reply:
Good point!